### PR TITLE
Fixes #2002. Added feature to fill the remaining width with spaces.

### LINKED ISSUE #2002: TextFormatter only draw the text width without fill the remaining bounds width with spaces.
If a view width is bigger than the text length and we needed to fill the remaining with spaces. This allows any view to leverage all the text alignment feature.
Another advantage of this fill option is that isn't needed to clear the area before write. The View.Redraw currently use two methods to write to the screen, Clear and Draw. Using only the Draw method is possible to clear what is to be cleared and write what have to be write, increasing the performance.

### PR DESCRIPTION
**Pull request by BDisp**
Fixes #2002 - By default the `TextFormatter.Draw` method should fill the remaining view width spaces. This way any view can fill all the view width. If it's false only the text width will be draw.

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [x] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

File: Terminal.Gui/Core/TextFormatter.cs [hunk 1]
```diff
@@ -1141,7 +1141,8 @@ public static ustring RemoveHotKeySpecifier (ustring text, int hotPos, Rune hotK
 		/// <param name="normalColor">The color to use for all text except the hotkey</param>
 		/// <param name="hotColor">The color to use to draw the hotkey</param>
 		/// <param name="containerBounds">Specifies the screen-relative location and maximum container size.</param>
-		public void Draw (Rect bounds, Attribute normalColor, Attribute hotColor, Rect containerBounds = default)
+		/// <param name="fillRemaining">Determines if the bounds width will be used (default)or only the text width will be used.</param>
```
Comment (tig) [inline Terminal.Gui/Core/TextFormatter.cs:1144]: You're missing a space after `(default)`.

**Review by tig — CHANGES_REQUESTED** [review]
I do not understand why this feature is needed.

The issue (#2002) does not describe the circumstances where the issue exists. 

Can you please elaborate?

**Comment by BDisp** [comment]
> I do not understand why this feature is needed.
> 
> The issue (#2002) does not describe the circumstances where the issue exists.
> 
> Can you please elaborate?

In the unit test already explain why this change is needed. If the view width is bigger than the text length and we needed to fill the remaining with spaces. This allows any view to leverage all the text alignment feature.

File: Terminal.Gui/Core/TextFormatter.cs (hunk 1 above)
Comment (BDisp) [reply Terminal.Gui/Core/TextFormatter.cs:1144]: I'll fix it, thanks.

File: UnitTests/TextFormatterTests.cs [hunk 2]
```diff
@@ -3334,6 +3334,35 @@ public void Draw_Vertical_Wide_TextAlignments ()
...
+			tf.Draw (new Rect (0, 0, 10, 1), view.GetNormalColor (), view.ColorScheme.HotNormal, default, false);
+			Application.Top.Redraw (Application.Top.Bounds);
+			GraphViewTests.AssertDriverContentsWithFrameAre (@"
+Helloy long text behind the TextFormatter.
+", output);
+
+			tf.Draw (new Rect (0, 0, 10, 1), view.GetNormalColor (), view.ColorScheme.HotNormal);
+			Application.Top.Redraw (Application.Top.Bounds);
+			GraphViewTests.AssertDriverContentsWithFrameAre (@"
+Hello     g text behind the TextFormatter.
```
Comment (tig) [inline UnitTests/TextFormatterTests.cs:3362]: I don't understand why anyone would want "A very long text behind the TextFormatter." to be formatted as "Hello     g text behind the TextFormatter."

File: UnitTests/TextFormatterTests.cs [hunk 3]
```diff
@@ -3334,6 +3334,35 @@ public void Draw_Vertical_Wide_TextAlignments ()
...
+			Application.Begin (Application.Top);
+
+			GraphViewTests.AssertDriverContentsWithFrameAre (@"
+A very long text behind the TextFormatter.
+", output);
+
+			tf.Draw (new Rect (0, 0, 10, 1), view.GetNormalColor (), view.ColorScheme.HotNormal, default, false);
+			Application.Top.Redraw (Application.Top.Bounds);
+			GraphViewTests.AssertDriverContentsWithFrameAre (@"
+Helloy long text behind the TextFormatter.
```
Comment (tig) [inline UnitTests/TextFormatterTests.cs:3356]: Why is this "Helloy " (with a `y`)??

Comment (BDisp) [reply UnitTests/TextFormatterTests.cs:3356]: The y belongs to the text beyond the Hello.

File: UnitTests/TextFormatterTests.cs (hunk 2 above)
Comment (BDisp) [reply UnitTests/TextFormatterTests.cs:3362]: I know it's only to explain that if we want the view width been respected the remaining will be filled by spaces. Imagine we want draw listview lines with it and leverage the alignment features.

File: UnitTests/TextFormatterTests.cs (hunk 3 above)
Comment (BDisp) [reply UnitTests/TextFormatterTests.cs:3356]: In this case the fill option wasn't used.

File: UnitTests/TextFormatterTests.cs (hunk 2 above)
Comment (BDisp) [reply UnitTests/TextFormatterTests.cs:3362]: Well that was what comes up on my mind at the time. What do you suggest to change this test?

File: Terminal.Gui/Core/TextFormatter.cs (hunk 1 above)
Comment (BDisp) [reply Terminal.Gui/Core/TextFormatter.cs]: Done @tig. Thanks for caught that.

File: UnitTests/TextFormatterTests.cs (hunk 2 above)
Comment (BDisp) [reply UnitTests/TextFormatterTests.cs]: I rewritten the unit test and I hope you can understand the main reason of this PR. With the `tf2` I cleared and written at once.

File: UnitTests/TextFormatterTests.cs (hunk 3 above)
Comment (BDisp) [reply UnitTests/TextFormatterTests.cs]: I rewritten the unit test. Thanks.

**Review by tig — APPROVED** [review]

**Pull request merged**